src/zcbor_decode.c

```diff
@@ -1008,12 +1012,15 @@ bool zcbor_any_skip(zcbor_state_t *state, void *result)
 		case ZCBOR_MAJOR_TYPE_BSTR:
 		case ZCBOR_MAJOR_TYPE_TSTR:
 			/* 'value' is the length of the BSTR or TSTR */
-			if (value > (state->payload_end - state->payload)) {
+			if (value > (size_t)(state->payload_end - state->payload)) {
```

Comment:
Should we have ZCBOR_CHECK_PAYLOAD(); here ?

Reply:
It happens in value_extract() above. Added a comment.